src/coreclr/jit/helperexpansion.cpp

```diff
@@ -413,6 +413,388 @@ bool Compiler::fgExpandRuntimeLookupsForCall(BasicBlock** pBlock, Statement* stm
+    }
+
+    if (!noSizeCheck)
+    {
+        GenTree* lengthCheck = gtNewOperNode(GT_LT, TYP_INT, gtClone(dstLen), srcLenCnsNode);
+        lengthCheck->gtFlags |= GTF_RELOP_JMP_USED;
+        Statement* lengthCheckStmt = fgNewStmtFromTree(gtNewOperNode(GT_JTRUE, TYP_VOID, lengthCheck), debugInfo);
+        fgInsertStmtAtEnd(lengthCheckBb, lengthCheckStmt);
+        lengthCheckBb->bbCodeOffs    = block->bbCodeOffsEnd;
+        lengthCheckBb->bbCodeOffsEnd = block->bbCodeOffsEnd;
+    }
+
+    //
+    // Block 2: fastpathBb - unrolled loop that copies the UTF8 const data to the destination
+    //
+    // We're going to emit a series of loads and stores to copy the data.
+    // In theory, we could just emit the const U8 data to the data section and use GT_BLK here
+    // but that would be a bit less efficient since we would have to load the data from memory.
+    //
+    BasicBlock* fastpathBb = fgNewBBafter(BBJ_NONE, lengthCheckBb, true);
+    fastpathBb->bbFlags |= BBF_INTERNAL;
+
+    // The widest type we can use for loads
+    const var_types maxLoadType = roundDownMaxRegSize(srcLenCns);
+
+    // How many iterations we need to copy UTF8 const data to the destination
+    unsigned iterations = srcLenCns / genTypeSize(maxLoadType);
+
+    // Add one more iteration if we have a remainder
+    iterations += srcLenCns % genTypeSize(maxLoadType) == 0 ? 0 : 1;
+
+    for (unsigned i = 0; i < iterations; i++)
+    {
+        ssize_t offest = (ssize_t)i * genTypeSize(maxLoadType);
+
+        // Last iteration: overlap with previous load if needed
+        if (i == iterations - 1)
+        {
+            offest = (ssize_t)srcLenCns - genTypeSize(maxLoadType);
+        }
+
+        // We're going to emit the following tree:
+
+        // -A-XG------       *  ASG       %maxLoadType% (copy)
+        // D--XG--N---       +--*  IND       %maxLoadType%
+        // -------N---       |  \--*  ADD       byref
+        // -----------       |     +--*  LCL_VAR   byref dstPtr
+        // -----------       |     \--*  CNS_INT   int   %offset%
+        // -----------       \--*  CNS_VEC or CNS_INT representing UTF8 const data chunk
+
+        GenTreeIntCon* offsetNode = gtNewIconNode(offest);
+        fgUpdateConstTreeValueNumber(offsetNode);
+
+        // Grab a chunk from srcUtf8cnsData for the given offset and width
+        GenTree* utf8cnsChunkNode = gtNewCon(maxLoadType, srcUtf8cns + offest);
+        fgUpdateConstTreeValueNumber(utf8cnsChunkNode);
+
+        GenTree*   dstAddOffsetNode = gtNewOperNode(GT_ADD, TYP_BYREF, gtClone(dstPtr), offsetNode);
+        GenTree*   indirNode        = gtNewIndir(maxLoadType, dstAddOffsetNode);
+        GenTreeOp* storeInd         = gtNewAssignNode(indirNode, utf8cnsChunkNode);
+        Statement* storeIndStmt     = fgNewStmtFromTree(storeInd, debugInfo);
+        fgInsertStmtAtEnd(fastpathBb, storeIndStmt);
+        gtUpdateStmtSideEffects(storeIndStmt);
```

Comment:
```suggestion
```

Don't think this is necessary, the nodes should propagate flags as they are created.